src/qt/test/proposaltests.cpp

```diff
@@ -0,0 +1,21 @@
+#include "proposaltests.h"
+#include <test/setup_common.h>
+
+#include <qt/governancelist.h>
+
+void ProposalTests::proposaltests()
+{
+    CGovernanceObject *pGovObj = new CGovernanceObject;
+    Proposal proposal(pGovObj);
+    proposal.m_currentDate = QDateTime(QDate(2022, 1, 2), QTime(12, 00));
+    proposal.m_endDate = QDateTime(QDate(2022, 2, 9), QTime(12, 00));
+    QVERIFY( proposal.paymentRemaining() == 1) ;
+
+    proposal.m_currentDate = QDateTime(QDate(2022, 1, 2), QTime(12, 00));
+    proposal.m_endDate = QDateTime(QDate(2022, 3, 5), QTime(12, 00));
+    QVERIFY( proposal.paymentRemaining() == 2);
+
+    proposal.m_currentDate = QDateTime(QDate(2022, 1, 2), QTime(12, 00));
+    proposal.m_endDate = QDateTime(QDate(2022, 4, 10), QTime(12, 00));
+    QVERIFY( proposal.paymentRemaining() == 3);
```

Comment:
should test the exact edges, and preferably de-duplicate this code smth like
```
const auto& check_paymentRemaining = [&proposal](const auto& begin, const auto& end, int expected) {
    proposal.m_currentDate = begin;
    proposal.m_endDate = end;
    QVERIFY( proposal.paymentRemaining() == expected) ;
}
```

Also, I think there may be a bug here... the currentDate and endDate shouldn't be enough to calculate payments remaining.. I think you also need to know when the last superblock was. @UdjinM6 maybe @thephez, what should the calculation be to determine payments remaining?

Reply:
Yes, it doesn't seem possible to fully evaluate without comparing against superblock times. I'm not sure what the full calculation would be, but it should be available somewhere in either Core or [Sentinel](https://github.com/dashpay/sentinel). Perhaps @UdjinM6 has a reference? :crossed_fingers: DMT definitely does this calc also since it displays this info. My guess would be that it happens somewhere in [here](https://github.com/Bertrand256/dash-masternode-tool/blob/master/src/proposals_dlg.py), but that's a long file.

Also, since the proposal start/end are defined as timestamps while the superblocks are based on block heights, it's not a precise calculation (i.e. it's impacted by variations in block time).